daffodil-core/src/main/scala/org/apache/daffodil/runtime1/SequenceTermRuntime1Mixin.scala

```diff
@@ -57,7 +57,7 @@ trait ChoiceBranchImpliedSequenceRuntime1Mixin { self: ChoiceBranchImpliedSequen
   override lazy val sequenceRuntimeData: SequenceRuntimeData = {
     new SequenceRuntimeData(
       position,
-      partialNextElementResolver,
+      Delay(partialNextElementResolver),
```

Comment:
Is there anything that forces these Delays here to be evaluated?

Reply:
All are now evaluated. They are forced because ModelGroup has a runtime mixin, and that has a requiredEvaluation that demands modelGroupRuntimeData.initialize, and that initialize method is responsible to demand all delayed values. 

This is the pattern it all follows now. If you are doing functional graph creation with cycles, then there will be an initialize method which is responsible for forcing the delays so you have the graph fully formed.